xarray/core/rolling.py

```diff
@@ -228,6 +228,13 @@ def reduce(self, func, **kwargs):
         reduced : DataArray
             Array with summarized data.
         """
+        # Reduce functions usually assumes numeric type.
+        # For non-number array such as bool, We cast them to float
+        if self.obj.dtype.kind not in 'iufcm':
+            return DataArrayRolling(
+                self.obj.astype(float), center=self.center,
+                min_periods=self.min_periods,
+                **{self.dim: self.window}).reduce(func, **kwargs)
```

Comment:
I think boolean array rolling should work, such as
```python
array.isnull().rolling(dim=1).sum()
```
We are actually using this [here](https://github.com/pydata/xarray/blob/e31cf43e8d183c63474b2898a0776fda72abc82c/xarray/core/missing.py#L332).

So I cast non-number array to `float` here.